### PR TITLE
Added pedantic file

### PR DESCRIPTION
**Pull request by Samdid**
## Related Issue 

- add pedantic file - I have to add a file containing all the rules

Closes: #67 

### Describe the changes you've made
I have added an "analysis_options.yaml" file which contains all the rules in pedantic dependency. 
## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Mention any unusual behaviour of your code (Write NA if not)
Any unusual behaviour of your code
NA

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly whereever it was hard to understand.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Additional Info (optional)
Any additional information you want to give
None

### CODE REVIEW

**Comment by Samdid** [comment]
@hiashutoshsingh please review this PR

File: pubspec.lock [hunk 1]
```diff
@@ -0,0 +1,1034 @@
+# Generated by pub
+# See https://dart.dev/tools/pub/glossary#lockfile
```
Comment (hiashutoshsingh) [inline pubspec.lock:2]: Please remove this lock file from commit, but keep this in your local.
See Why: https://dart.dev/guides/libraries/private-files#pubspeclock

**Review by hiashutoshsingh — CHANGES_REQUESTED** [review]
This is seriously good work. 👏👏👏
It just a step in achieving a good quality of code.


Just a small change needed, please remove the lock file.

File: pubspec.lock (hunk 1 above)
Comment (Samdid) [reply pubspec.lock:2]: Okay, I will remove that

**Comment by Samdid** [comment]
@hiashutoshsingh Resolved the changes

**Review by hiashutoshsingh — APPROVED** [review]
Thanks for the contribution.

**Pull request merged**